### PR TITLE
tests: Add mocks and tests for libvirt backend driver

### LINKED ISSUE #410: Put libvirt backend under test
The libvirt backend needs to be put under testing for better coverage.

### PR DESCRIPTION
**Pull request by townsend2010**
Fixes #410

### CODE REVIEW

File: tests/test_libvirt_backend.cpp [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+
+TEST_F(LibVirtBackend, failed_connection_throws)
+{
+    REPLACE(virConnectOpen, [](auto...) { return nullptr; });
+    EXPECT_THROW(mp::LibVirtVirtualMachineFactory backend{data_dir.path()}, std::runtime_error);
+}
+
+TEST_F(LibVirtBackend, creates_in_off_state)
+{
+    REPLACE(virConnectOpen, [](auto...) { return reinterpret_cast<virConnectPtr>(0x1); });
```
Comment (albaguirre) [inline tests/test_libvirt_backend.cpp]: Maybe all the reinterpret_cast calls can be turned into return fake_handle<T>();

where fake_handle is:

```
namespace
{
template<typename T>
auto fake_handle()
{
    return reinterpret_cast<T>(0xDEADBEEF);
}
}
```

**Comment by Saviq** [comment]
Is the Mac conflict expected?

**Comment by townsend2010** [comment]
@Saviq,

Yeah, I'm wondering about the Mac conflict too. I'll investigate that.

**Comment by townsend2010** [comment]
Ok, the Mac conflict will have to be fixed by doing a merge into full.

**Comment by townsend2010** [comment]
Not going to bother with bors since the Mac build is failing.  Will merge manually instead.

**Pull request merged**